app/javascript/users/UserEdit.re

```diff
@@ -0,0 +1,584 @@
+    let isInvalidImageFile =
+      file##size > maxAllowedFileSize
+      || (
+        switch (file##_type) {
+        | "image/jpeg"
+        | "image/gif"
+        | "image/png" => false
+        | _ => true
+        }
+      );
+
+    let error =
+      isInvalidImageFile
+        ? Some("Please select an image with a size less than 5 MB") : None;
+
+    switch (error) {
+    | Some(error) => send(SetAvatarUploadError(Some(error)))
+    | None => submitAvatarForm(send, formId)
+    };
+  };
+};
+
+let updateUser = (state, send, id, event) => {
+  ReactEvent.Mouse.preventDefault(event);
+  send(StartSaving);
+
+  UpdateUserQuery.make(
+    ~id,
+    ~name=state.name,
+    ~about=state.about,
+    ~currentPassword=state.currentPassword,
+    ~newPassword=state.newPassword,
+    ~confirmPassword=state.confirmPassword,
+    ~dailyDigest=state.dailyDigest,
+    (),
+  )
+  |> GraphqlQuery.sendQuery
+  |> Js.Promise.then_(result => {
+       result##updateUser##success
+         ? {
+           let hasCurrentPassword = state.newPassword |> String.length > 0;
+           send(FinishSaving(hasCurrentPassword));
+         }
+         : send(FinishSaving(state.hasCurrentPassword));
+       Js.Promise.resolve();
+     })
+  |> Js.Promise.catch(_ => {
+       send(FinishSaving(state.hasCurrentPassword));
+       Js.Promise.resolve();
+     })
+  |> ignore;
+  ();
+};
+
+let initiateAccountDeletion = (state, send, id) => {
+  send(StartDeletingAccount);
+
+  InitiateAccountDeletionQuery.make(
+    ~id,
+    ~password=state.passwordForAccountDeletion,
+    (),
+  )
+  |> GraphqlQuery.sendQuery
+  |> Js.Promise.then_(result => {
+       result##initiateAccountDeletion##success
+         ? {
+           send(FinishAccountDeletion);
+         }
+         : send(FinishAccountDeletion);
+       Js.Promise.resolve();
+     })
+  |> Js.Promise.catch(_ => {
+       send(FinishAccountDeletion);
+       Js.Promise.resolve();
+     })
+  |> ignore;
+  ();
+};
+
+let hasInvalidPassword = state => {
+  state.newPassword == ""
+  && state.confirmPassword == ""
+  || state.newPassword == state.confirmPassword
+  && state.newPassword
+  |> String.length >= 8
+    ? false : true;
+};
+
+let saveDisabled = state => {
+  hasInvalidPassword(state)
+  || state.name
+  |> String.trim
+  |> String.length < 2
+  || !state.dirty;
+};
+
+let confirmDeletionWindow = (state, send, currentUserId) => {
+  state.showDeleteAccountForm
+    ? {
+      let body =
+        <div ariaLabel="Confirm dialog for account deletion">
+          <p className="text-sm text-center sm:text-left text-gray-700">
+            {"Are you sure you want to delete your account? All of your data will be permanently removed from our servers forever. This action cannot be undone."
+             |> str}
+          </p>
+          <div className="mt-3">
+            <label htmlFor="password" className="block text-sm font-semibold">
+              {"Password" |> str}
+            </label>
```

Comment:
As discussed, we'll need to switch to accepting the email address of the signed in user.